webapp/channels/src/components/about_build_modal/about_build_modal.tsx

```diff
@@ -55,6 +62,11 @@ export default class AboutBuildModal extends React.PureComponent<Props, State> {
         const config = this.props.config;
         const license = this.props.license;
 
+        var serverHostname = 'disconnected';
+        if (this.props.socketStatus.connected) {
+            serverHostname = this.props.socketStatus.serverHostname ?? 'unknown';
+        }
```

Comment:
These strings won't be localized with the current implementation - I was unsure how important that would be for this feature vs. maintaining these in other languages.

Reply:
We will need to translate these strings, we only need to setup the English translation and the rest will be taken care of separately by our translation teams. `FormattedMessage` doesn't quite work in this situation because you need a string and `useIntl` is a hook which only works in functional components. I recommend we convert this component to a functional component and use `useIntl` from `react-intl` for the translation. We are actively trying to move away from class based components anyway so this will be a necessary task at some point in the future anyway.

Reply:
It's verbose, but we could leverage `FormattedMessage` with something like:

    let serverHostname;
    if (!this.props.socketStatus.connected) {
        serverHostname = (
            <div>
                <FormattedMessage
                    id='about.serverHostname'
                    defaultMessage='Hostname:'
                />
                <Nbsp/>
                <FormattedMessage
                    id='about.serverDisconnected'
                    defaultMessage='disconnected'
                />
            </div>
        );
    } else if (this.props.socketStatus.serverHostname) {
        serverHostname = (
            <div>
                <FormattedMessage
                    id='about.serverHostname'
                    defaultMessage='Hostname:'
                />
                <Nbsp/>
                {this.props.socketStatus.serverHostname}
            </div>
        );
    } else {
        serverHostname = (
            <div>
                <FormattedMessage
                    id='about.serverHostname'
                    defaultMessage='Hostname:'
                />
                <Nbsp/>
                <FormattedMessage
                    id='about.serverUnknown'
                    defaultMessage='unknown'
                />
            </div>
        );
    }

I think I would prefer this approach for now, and we could do the `class -> function` conversion in a separate task? I'm worried doing that change here will add a lot of scope to this PR. WDYT?

Reply:
Yeah that works for me, my only thought is whether `Hostname: disconnected` and `Hostname: unknown` should be single translation strings or if it's ok from a translation perspective to have `unknown` and `disconnected` as their own translations. I don't have answer for that so it might be worth checking with @cwarnermm

Reply:
Now that you ask and thinking about this more, I'm wondering about these two strings, too. I think they should be separate: a disconnected websocket is a normal condition that a user can get into, but an unknown host is a failure condition where something is wrong on the server (e.g. with hostname configuration). 

Instead of reporting `unknown` for the latter, maybe we should have an explicit error string? Something like: `server unable to resolve hostname` or similar.

Reply:
Yeah I think an explicit error string instead of unknown makes sense tbh

Reply:
Great discussion; I'm 5/5 with decisions made.

Reply:
Really great discussion! I'm also 5/5 on decisions made.